knowledge-base/display-custom-error-messages-on-web-based-viewer.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Display custom error messages on web-based viewer
+description: This article elaborates on how to replace any error message output by the report with a custom error message.
+type: how-to
+page_title: Display custom error messages on web-based viewer
+slug: display-custom-error-messages-on-web-based-viewer
+position: 
+tags: HTML5Viewer, Localization
+ticketid: 1410107
+res_type: kb
+---
+
+## Environment
+<table>
+    <tbody>
+	    <tr>
+	    	<td>Product</td>
+	    	<td>Progress® Telerik® Reporting</td>
+	    </tr>
+	    <tr>
+	    	<td>Report Viewer</td>
+	    	<td>All web-based viewers</td>
+	    </tr>
+    </tbody>
+</table>
+
+
+## Description
+How do I change the default error messages displayed by the HTML5-based Report Viewers with my custom ones? Example of such a message is the following:
+
+```
+Unable to get report parameters.
+An error has occurred.
+An error has occurred while resolving X data source: An error occurred while invoking data retrieval method.
+```
+
+## Solution
+Messages displayed by the HTML5-based Report Viewers are separate into two "categories". While the string "Unable to get report parameters." is a client-side message, the rest of the message is server-side one.
+
+To change the client side text, you might use the approach demonstrated in the [Localization](../html5-report-viewer-localization) article and change the corresponding value. However, this would not remove the server-side message. To change the entire message, the following code snippet could be used:
```

Comment:
By the new style guide "could" should be avoided. You can use "To change the entire message, use the following code snippet:"